File: android/src/main/java/com/giphyreactnativesdk/RTNGiphySdkInfo.kt
package com.giphyreactnativesdk

import android.content.Context
import android.content.pm.PackageInfo
import android.content.pm.PackageManager

class RTNGiphySdkInfo(val context: Context) {
  val name = "RNSDK"

  val version: String
    get() {
      var packageInfo: PackageInfo? = null
      try {
        packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
      } catch (e: PackageManager.NameNotFoundException) {
        e.printStackTrace()
      }

      return packageInfo?.versionName.toString()
    }
}

File: ios/Utils/RTNGiphySDKInfo.swift
class RTNGiphySDKInfo {
  static let shared = RTNGiphySDKInfo()

  let name = "RNSDK"

  lazy var version: String? = {
    if let rawVersion = Bundle(for: type(of: self)).object(forInfoDictionaryKey: "CFBundleShortVersionString") {
      return rawVersion as? String
    }
    return nil
  }()
}
